### ISSUE TITLE
NullPointerException when using verbose reporter and data provider with null values 

### ISSUE THREAD

**Issue by ipotapchuk** [cbeust/testng#2644]
### TestNG Version
7.4.0

### Expected behavior
No NullPointerException when using null values in data provider and verbose reporter

### Actual behavior
NullPointerException when using null values in data provider and verbose reporter:
```
java.lang.NullPointerException
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:550)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at org.testng.internal.Utils.extractParameterTypes(Utils.java:612)
	at org.testng.reporters.VerboseReporter.getMethodDeclaration(VerboseReporter.java:268)
	at org.testng.reporters.VerboseReporter.logTestResult(VerboseReporter.java:185)
	at org.testng.reporters.VerboseReporter.onTestFailure(VerboseReporter.java:78)
	at org.testng.internal.TestListenerHelper.runTestListeners(TestListenerHelper.java:96)
	at org.testng.internal.TestInvoker.runTestResultListener(TestInvoker.java:219)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:656)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:173)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:824)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:146)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:794)
	at org.testng.TestRunner.run(TestRunner.java:596)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
	at org.testng.SuiteRunner.run(SuiteRunner.java:276)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
	at org.testng.TestNG.runSuites(TestNG.java:1063)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:99)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

### Is the issue reproducible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
Test example:
```java
public class NullDataProviderValueWithVerboseReporterTest {

	@Test(dataProvider = "nullProvider")
	public void test(String value) {

	}

	@DataProvider
	public Object[][] nullProvider() {
		return new Object[][] {
				{ null }
		};
	}
}
```

Failsafe configuration (in pom):
```xml
<plugin>
	<groupId>org.apache.maven.plugins</groupId>
	<artifactId>maven-failsafe-plugin</artifactId>
	<version>3.0.0-M3</version>
	<configuration>
		<properties>
			<property>
				<name>surefire.testng.verbose</name>
				<value>${testng.verbose.level}</value>
		        </property>
		</properties>
	</configuration>
        <executions>
		<execution>
			<goals>
				<goal>integration-test</goal>
				<goal>verify</goal>
			</goals>
		</execution>
	</executions>
</plugin>
```

Run with command:
```bash
mvn -Dtestng.verbose.level=10 -Dit.test=NullDataProviderValueWithVerboseReporterTest -f pom.xml verify
```

**Comment by krmahadevan** [comment]
This issue is fixed already in the latest snapshot version of TestNG and the fix should be available once `7.5.0` is released.

To confirm, you can do the following:

Add the below repository tag to your pom file

```xml
<repository>
    <id>snapshots-repo</id>
    <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    <releases>
        <enabled>false</enabled>
    </releases>
    <snapshots>
        <enabled>true</enabled>
    </snapshots>
</repository>
```

And add a dependency on TestNG snapshot version as below

```xml
<dependency>
    <groupId>org.testng</groupId>
    <artifactId>testng</artifactId>
    <version>7.5-SNAPSHOT</version>
</dependency>
```

**Comment by dankirkd** [comment]
Any idea when 7.5.0 will be released?

**Comment by dankirkd** [comment]
@krmahadevan Any idea when 7.5.0 will be released so we can get this fix?

**Comment by krmahadevan** [comment]
@dankirkd - Short answer: I dont have a definite timeline.

Long answer: We are trying to fix one bug that was introduced in `7.5.0` as a result of us trying to fix another bug. Now this issue seems to be like a blocker for a team who are trying to get past this so that they can take the upgrade as well. We are collaborating on the fix and as soon as the fix is available `7.5.0` should be released.

The PR in question : https://github.com/cbeust/testng/pull/2666

@juherr - Would you want to add anything additional ?

**Comment by dankirkd** [comment]
@krmahadevan Thanks.  I can see why the issue you point to isn't a feature that many folks are reporting issues with given the conventional wisdom tends to steer away from writing tests that have to run in a given order, as defined perhaps using DependsOnGroups.  I've also seen the discussion about more frequent releases, which helps in these situations.
